File: Jihun/week_8/2841.py
import sys
input = sys.stdin.readline

N, P = map(int, input().split())

stack = [[0] for i in range(7)]
count = 0

for i in range(N):
    n, p = map(int, input().split())
    if stack[n][-1] < p:
        stack[n].append(p)
        count += 1
    elif stack[n][-1] > p:
        while len(stack[n]) >= 1:
            stack[n].pop()
            count += 1

            if stack[n][-1] == p:
                break
            if stack[n][-1] < p:
                count += 1
                stack[n].append(p)
                break
print(count)

File: JongBin/8week/2841.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;
import java.util.StringTokenizer;

class Main{
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(st.nextToken());
        int P = Integer.parseInt(st.nextToken());

        Stack<Integer>[] stacks = new Stack[N];
        for(int i =0; i<N; i++){
            stacks[i] = new Stack<>();
        }
        int count = 0;
        for(int i =0; i<N; i++){
            st = new StringTokenizer(br.readLine());
            int A = Integer.parseInt(st.nextToken());
            int B = Integer.parseInt(st.nextToken());

            if(stacks[A].isEmpty() || stacks[A].peek()<B){
                //stack이 비었을때
            }else if(stacks[A].peek()>B){
                while(!stacks[A].isEmpty() && stacks[A].peek()>B){
                    stacks[A].pop();
                    count++;
                }
                if(!stacks[A].isEmpty() && stacks[A].peek() == B){
                    continue;
                }
            }else{
                continue;
            }
            stacks[A].push(B);

            count++;
        }
        System.out.println(count);
    }
}